runtime/src/lib.rs

```diff
@@ -257,11 +257,11 @@ pub const WEIGHT_PER_GAS: u64 = WEIGHT_PER_SECOND / GAS_PER_SECOND;
 pub struct MoonbeamGasWeightMapping;
 
 impl pallet_evm::GasWeightMapping for MoonbeamGasWeightMapping {
-	fn gas_to_weight(gas: usize) -> Weight {
+	fn gas_to_weight(gas: u64) -> Weight {
 		Weight::try_from((gas as u64).saturating_mul(WEIGHT_PER_GAS)).unwrap_or(Weight::MAX)
```

Comment:
Yeah finally a correct type for gas.
You can remove the `as 64` here and maybe most of the rest too as both the Weight and the gas are 64 I think

Reply:
I kept `u32::MAX as u64` in hopes of being consistent with our previous fallback. Is this ever useful, anyway, and would u64::MAX somehow be better? (or were you merely suggesting that 'as u64' was redundant?)